peekaboo/toolbox/file.py

```diff
@@ -0,0 +1,116 @@
+# your option) any later version.                                             #
+#                                                                             #
+# This program is distributed in the hope that it will be useful, but         #
+# WITHOUT ANY WARRANTY; without even the implied warranty of                  #
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU           #
+# General Public License for more details.                                    #
+#                                                                             #
+# You should have received a copy of the GNU General Public License           #
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.       #
+#                                                                             #
+###############################################################################
+
+
+import logging
+import mimetypes
+import magic
+
+
+logger = logging.getLogger(__name__)
+
+
+class Filetools(object):
+    """ Parent class, defines interface to various file tools. """
+
+    def guess_mime_type_from_contents(self, file_path):
+        """  Get type from file magic bytes. """
+        mime_types = magic.from_file(file_path, mime=True)
+        if not mime_types:
+            return None
+
+        return mime_types
+
+    def guess_mime_type_from_name(self, file_path):
+        """ Guess the type of a file based on its filename or URL. """
+        if not mimetypes.inited:
+            mimetypes.init()
+            mimetypes.add_type('application/javascript', '.jse')
+
+        mime_types = mimetypes.guess_type(file_path)[0]
+        if not mime_types:
+            return None
+
+        return mime_types
+
+    def guess_mime_type_text_representation(self, file_path):
+        """ Guess the type by content and hand back text representation rather than
+        mime type. """
+        type_as_text = magic.from_file(file_path)
+        if not type_as_text:
+            return None
+
+        return type_as_text
+
+    def get_report(self, sample):
+        """ Return filetools report or create if not already cached. """
+        report = {}
+
+        report["type_by_content"] = self.guess_mime_type_from_contents(sample.file_path)
+        report["type_by_name"] = self.guess_mime_type_from_name(sample.filename)
+        report["type_as_text"] = self.guess_mime_type_text_representation(sample.file_path)
+
+        ftreport = FiletoolsReport(report)
+        sample.register_filetools_report(ftreport)
+        return ftreport
+
+
+class FiletoolsReport(object):
+    """ Represents a custom Filetools report. """
+    def __init__(self, report):
+        self.report = report
+
+    def __str__(self):
+        return "<FiletoolsReport('%s'>" % self.report
+
+    @property
+    def type_by_content(self):
+        """
+        Guesses the type of a file based on the files contents.
```

Comment:
Phrasing of comments: These are passive properties and don't do anything themselves but just reflect what's been guessed/determined/done by the analyser. Could be rephrased like "The type of a file guessed based on ...".

Reply:
Addressed in #143.